### PR TITLE
Api ipc

### PR DESCRIPTION
**Pull request by Mingling94**

### CODE REVIEW

**Comment by DavidVorick** [comment]
I know this might be a PITA, but we should probably have all variables dealing with siacoin volumes be infinite precision.

File: app/dependencies/siamath.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	return string + " KS";
+}
+
+// precision determines the length of number
+function limitPrecision(number, precision) {
+	return number.toString().slice(0,precision+1);
+}
+
+// engineering notation
+function engNotation(number, precision) {
```
Comment (DavidVorick) [inline app/dependencies/siamath.js:38]: probably not needed, guess it doesn't hurt to have

File: app/dependencies/siamath.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+// Utility file to provide math functions
+'use strict';
+
+var pSI = ["", "k", "M", "G", "T"];
+var nSI = ["", "m", "&micro;", "&nano;", "&pico;"];
+var siaConversionFactor = Math.pow(10,27);
```
Comment (DavidVorick) [inline app/dependencies/siamath.js:6]: siaConversionFactor is 10^24

File: app/dependencies/siamath.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	return mcoin / siaConversionFactor;
+}
+
+// siacoin to base units
+function baseUnit(units) {
+	return units * siaConversionFactor;
+}
+
+// base units & precision to siacoin
+function fksiacoin(mcoin, l) {
```
Comment (DavidVorick) [inline app/dependencies/siamath.js:19]: I don't understand this function name

File: app/dependencies/siamath.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
+// Utility file to provide math functions
+'use strict';
+
+var pSI = ["", "k", "M", "G", "T"];
+var nSI = ["", "m", "&micro;", "&nano;", "&pico;"];
+var siaConversionFactor = Math.pow(10,27);
+
+// base units to siacoin
+function siacoin(mcoin) {
```
Comment (DavidVorick) [inline app/dependencies/siamath.js:9]: mcoin is a bad variable name. it should probably be `baseUnits`

File: app/dependencies/siamath.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	return mcoin / siaConversionFactor;
+}
+
+// siacoin to base units
+function baseUnit(units) {
```
Comment (DavidVorick) [inline app/dependencies/siamath.js:14]: units is a bad variable name, it should be base units.

The function name also doesn't indicate that you are going from siacoins to base units. This function (and the above one) probably aren't necessary.

File: app/dependencies/siamath.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+			e -= 20;
+			x /= Math.pow(10,e);
+			x += (new Array(e+1)).join('0');
+		}
+	}
+	return x;
+}
+
+// properly controls data size representation
+function formatBytes(bytes) {
```
Comment (DavidVorick) [inline app/dependencies/siamath.js:85]: numBytes is probably a better variable name.

File: app/js/daemonAPI.js [hunk 7]
```diff
@@ -3,60 +3,86 @@
...
+		callback(new Error('Server not reached'));
+	};
+
+	// actually send the request
+	request.send(params);
+} 
+
+// discernCall() takes an unknown call object, and apropriately calls apiCall()
+// callback(err, result); if success, err is null
+function discernCall(call, callback) {
```
Comment (DavidVorick) [inline app/js/daemonAPI.js:39]: The docstring isn't very helpful.

File: app/js/plugin.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+		
+		// TODO: common css styling injected in plugins?
+		/*
+		Fs.readFile(Path.join(__dirname, '../css/general.css'), 'utf8', function(err, file) {
+			if (err) {
+				console.error(err);
+			}
+			view.insertCSS(file);
+		});
+	   */
```
Comment (DavidVorick) [inline app/js/plugin.js:78]: no need, injected CSS doesn't make a whole lot of sense here.

Copy-paste is simple enough for new devs, we want veteran devs to have full flexibility to manage their own CSS.

**Comment by DavidVorick** [comment]
This looks mostly alright. Admittedly, I don't understand it very well.

Tomorrow we'll be going over the whole thing, including the other pull request.

**Comment by Mingling94** [comment]
Hahahahhahahhahhahahhahhaha
The siamath file was pulled directly from siaui so blame the bad docstrings
and var names on Seve :p

Also ask @triazo about the fksiacoin function name

On Thu, Jul 16, 2015, 3:28 AM David Vorick notifications@github.com wrote:

> This looks mostly alright. Admittedly, I don't understand it very well.
> 
> Tomorrow we'll be going over the whole thing, including the other pull
> request.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NebulousLabs/test-ui/pull/16#issuecomment-121859079.
> 
> ## 
> 
> Sincerely and cheerily,

Ming Luo
Rensselaer Polytechnic Institute '15
B.S. Computer Science
Cell: (434) 420-2952
mcluo1994@gmail.com

**Comment by Mingling94** [comment]
Closing this PR because I'm building upon it in #17 and making the corrections there

**Pull request closed without merging**